### PR TITLE
#2605 Adapt Json Parsing for sslyze 6.0.0

### PR DESCRIPTION
**Pull request by BorisShek**
Upgrade sslyze to 6.0.0.
Update all JSON test files and parser.test.js to be compatible with sslyze 6.0.0 JSON output. Remove mixed-connectivity-result.json as it is no longer used in parser.test.js. Adjust JSON output for "self-signed.badssl.com" due to sslyze 6.0.0's migration to Python Cryptography for certificate validation. Validation errors for "self signed certificate" and "self signed certificate in certificate chain" now return as "validation failed: Other('Certificate is missing required extension')". Update the "matchesHostname" logic to align with sslyze 6.0.0's new validation approach.

### Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Make sure that all your commits are signed-off and that you are added to the [Contributors](https://github.com/secureCodeBox/secureCodeBox/blob/main/CONTRIBUTORS.md) file.
* [x] Make sure that all CI finish successfully.
* [x] Optional (but appreciated): Make sure that all commits are [Verified](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).

### CODE REVIEW

**Comment by BorisShek** [comment]
Resolves #2605

File: scanners/sslyze/parser/parser.js [hunk 1]
```diff
@@ -161,6 +161,8 @@ function analyseCertificateDeployments(serverScanResult) {
       return [];
     }
 
+    console.log(certificateInfos)
```
Comment (J12934) [inline scanners/sslyze/parser/parser.js:164]:
```suggestion
```

File: scanners/sslyze/parser/parser.js [hunk 2]
```diff
@@ -220,24 +222,32 @@ function analyseCertificateDeployment(certificateDeployment) {
...
     }
   }
 
-  const matchesHostname =
-    certificateDeployment.leaf_certificate_subject_matches_hostname;
+  // Access the leaf certificate in the chain
+  const leafCertificate = certificateDeployment.received_certificate_chain[0];
+
+  // Check if the certificate is self-signed by comparing subject and issuer
+  const isSelfSigned = leafCertificate.subject.rfc4514_string === leafCertificate.issuer.rfc4514_string;
```
Comment (J12934) [inline scanners/sslyze/parser/parser.js:236]: Ah cool you found a way to still have the self signed check :+1:

**Review by J12934 — CHANGES_REQUESTED** [review]
Awesome, looks good :)
Only one minor thing i noticed, otherwise it should be ready to go :+1:

**Review by J12934 — APPROVED** [review]

**Pull request merged**